### PR TITLE
fix secure profile with proxy online mode

### PR DESCRIPTION
**Pull request by NonSwag**
This pr is same as the last one in #9183
I messed up with the commit, so i made this one

### CODE REVIEW

**Review by Warriorrrr — CHANGES_REQUESTED** [review]
Seems like the code here is from an earlier state of the branch since https://github.com/PaperMC/Paper/pull/9183#pullrequestreview-1415895257 applies again.

**Comment by NonSwag** [comment]
Oh, sorry, that's my fault.
I forgot to check the commit.

**Comment by NonSwag** [comment]
Finally, rebased the code
totally forgot about this pr sorry

**Comment by NonSwag** [comment]
bump

File: patches/server/0986-fix-secure-profile-with-proxy-online-mode.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+index a7e133f3495e9132a5fdae2c24f225e7b026295a..03f895c65d3d516bfdc40ef323794d2d567ca594 100644
+--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
++++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+@@ -637,7 +637,10 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
+ 
+     @Override
+     public boolean enforceSecureProfile() {
+-        return this.getProperties().enforceSecureProfile && this.getProperties().onlineMode;
++        // Paper start - fix secure profile with proxy online mode
++        var globalConfiguration = io.papermc.paper.configuration.GlobalConfiguration.get();
```
Comment (Warriorrrr) [inline patches/server/0986-fix-secure-profile-with-proxy-online-mode.patch:17]: This globalConfiguration variable isn't needed

**Review by Warriorrrr — COMMENTED** [review]
Might be nice to put this behind a config option so that it isn't immediately enabled, proxies and chat signing is still a bit iffy

File: patches/server/0986-fix-secure-profile-with-proxy-online-mode.patch (hunk 1 above)
Comment (NonSwag) [reply patches/server/0986-fix-secure-profile-with-proxy-online-mode.patch:17]: What would you name the config entry for this?
And I am using this for a few months now and have no problems at all behind my velocity proxy (and without) and the chat is working as it should even with a custom chat format and (when modified) with modified messages.

**Comment by NonSwag** [comment]
Failed to check out the right branch... again...
I'll close this and move it to a new pr... again...

**Pull request closed without merging**